crates/bws/src/state.rs

```diff
@@ -0,0 +1,35 @@
+use std::path::PathBuf;
+
+use directories::BaseDirs;
+
+pub(crate) const BWS_DIRECTORY: &str = ".bws";
+pub(crate) const STATE_DIRECTORY: &str = "state";
+
+pub(crate) fn get_state_file_path(
+    state_file_dir: Option<PathBuf>,
+    access_token_id: String,
+) -> PathBuf {
+    let state_file_path = match state_file_dir {
+        Some(sfd) => {
+            let mut state_file_path = sfd;
```

Comment:
```suggestion
        Some(mut state_file_path) => {
```

Reply:
Resolved in [16703f4](https://github.com/bitwarden/sdk/pull/388/commits/16703f4eebf7cc48a1de24840e5ede3832864590)